### PR TITLE
Remove BigNumbers.js 

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-1803

Following the understanding of Bruno's PR (https://github.com/rainbow-me/rainbow/pull/3196), we don't need the BN precision in user-facing numbers and this is the motivation for his changes. I wanted to go a step further and in those specific cases just drop strings and use numbers in those places that are more efficient than numbers. 

This should not give any regression compared to Bruno's work (which was confirmed to not have regressions over `develop`) since all of the places were already using reduced precision anyway.

Additionally, I added a field `amountNum` to assets so we don't need to constantly traverse the same string to a number.

Also, I added similar things in savings for the same reason.  

I changed typings where needed. 

To observe the improvement I used:

```js
export function trackPerf(fn, name) {
  return function wrapper() {
    const t = performance.now();

    const res = fn.apply(this, arguments);

    const total = performance.now() - t;

    global.console.log(`Event ${name}: ${total.toFixed(2)}ms`); // global is used to see logs in production

    return res;
  };
}
```

And wrapped 
```js
trackPerf(sortAssetsByNativeAmount)
```

Before:
Event undefined 16 96ms

After:
LOG  43734 PM  Initialize pool positions

I observe a slight improvement in iOS (around 30-40% I guess)

## Dev checklist for QA: what to test

Compare numbers everywhere in the app.
IU expect we need a full regression for this

### CODE REVIEW

File: src/hooks/useSavingsAccount.js [hunk 1]
```diff
@@ -117,15 +117,31 @@ export default function useSavingsAccount(includeDefaultDai) {
...
+      ...resultData?.account,
+      tokens: resultData?.account?.tokens.map(token => ({
+        ...token,
+        lifetimeSupplyInterestAccruedNum: parseFloat(
+          token.lifetimeSupplyInterestAccrued
+        ),
+        supplyBalanceUnderlyingNum: parseFloat(token.supplyBalanceUnderlying),
+      })),
+    },
+  }));
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/hooks/useSavingsAccount.js:144]: yes! Previously we were parsing those strings around 1000 times. This is happening once

File: src/helpers/utilities.ts [hunk 2]
```diff
@@ -7,38 +7,71 @@ const stringify = (value: BigNumberish) => {
...
-export const add = (numberOne: BigNumberish, numberTwo: BigNumberish): string =>
-  stringify(
-    parseFloat(numberOne?.toString() || '0') +
-      parseFloat(numberTwo?.toString() || '0')
-  );
+export const add = (
+  numberOne: BigNumberish,
+  numberTwo: BigNumberish
+): number => {
+  if (typeof numberOne === 'number' && typeof numberTwo === 'number') {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/helpers/utilities.ts:14]: yea, I forgot to remove some loeftover from this file

File: src/helpers/buildWalletSections.js [hunk 3]
```diff
@@ -172,19 +172,16 @@ const withBriefUniswapSection = (
...
       underlyingBalanceNativeValue,
       underlyingPrice,
     } = asset;
-    totalUnderlyingNativeValue = add(
-      totalUnderlyingNativeValue,
-      underlyingBalanceNativeValue || 0
-    );
-    const lifetimeSupplyInterestAccruedNative = lifetimeSupplyInterestAccrued
-      ? multiply(lifetimeSupplyInterestAccrued, underlyingPrice)
+    totalUnderlyingNativeValue = underlyingBalanceNativeValue || 0;
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/helpers/buildWalletSections.js:182]: yea, but this doesn't make sense. 5 lines above we set `totalUnderlyingNativeValue` to 0

File: src/helpers/utilities.ts [hunk 4]
```diff
@@ -7,38 +7,71 @@ const stringify = (value: BigNumberish) => {
...
 
-export const add = (numberOne: BigNumberish, numberTwo: BigNumberish): string =>
-  stringify(
-    parseFloat(numberOne?.toString() || '0') +
-      parseFloat(numberTwo?.toString() || '0')
-  );
+export const add = (
+  numberOne: BigNumberish,
+  numberTwo: BigNumberish
+): number => {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/helpers/utilities.ts:13]: I checked a lot of places and this was ok. I think this is similar to tour change BigNumbers -> String

File: src/helpers/utilities.ts [hunk 5]
```diff
@@ -7,38 +7,71 @@ const stringify = (value: BigNumberish) => {
...
+  if (typeof numberOne === 'number' && typeof numberTwo === 'number') {
+    return numberOne + numberTwo;
+  }
+
+  const numberOneAsNumber =
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/helpers/utilities.ts:18]: I would avoid this

[the remark this answers is not in the document]
Comment (osdnk) [reply src/helpers/utilities.ts:14]: 2 reasons:
This is in total saving us maybe 5 lines.
If this is so much about optimization, I would avoid extra function call

**Comment by osdnk** [comment]
\rebase

File: src/handlers/ens.ts [hunk 6]
```diff
@@ -274,9 +274,10 @@ export const estimateENSRegistrationGasLimit = async (
   const registerWithConfigGasLimit = `${ethUnits.ens_register_with_config}`;
 
   const totalRegistrationGasLimit =
-    [...gasLimits, registerWithConfigGasLimit].reduce((a, b) =>
-      add(a || 0, b || 0)
-    ) || `${ethUnits.ens_registration}`;
+    [...gasLimits, registerWithConfigGasLimit].reduce(
+      (a, b) => add(a || 0, b || 0),
```
Comment (jkadamczyk) [inline src/handlers/ens.ts:278]: Why not use optional chaining here?

Comment (jkadamczyk) [inline src/handlers/ens.ts:278]: Why not use nullish coalescing here?

File: src/handlers/ens.ts [hunk 7]
```diff
@@ -333,7 +334,7 @@ export const estimateENSRegisterSetRecordsAndNameGasLimit = async ({
 
   const gasLimits = await Promise.all(promises);
 
-  const gasLimit = gasLimits.reduce((a, b) => add(a || 0, b || 0));
+  const gasLimit = gasLimits.reduce((a, b) => add(a || 0, b || 0), 0);
```
Comment (jkadamczyk) [inline src/handlers/ens.ts:337]: Same
Why not use nullish coalescing here?

File: src/helpers/buildWalletSections.js [hunk 8]
```diff
@@ -207,13 +204,10 @@ const withBalanceSavingsSection = (savings, network) => {
...
 const withBriefBalanceSavingsSection = (savings, isLoadingAssets, network) => {
-  let totalUnderlyingNativeValue = '0';
+  let totalUnderlyingNativeValue = 0;
   for (let saving of savings) {
     const { underlyingBalanceNativeValue } = saving;
-    totalUnderlyingNativeValue = add(
-      totalUnderlyingNativeValue,
-      underlyingBalanceNativeValue || 0
-    );
+    totalUnderlyingNativeValue = underlyingBalanceNativeValue || 0;
```
Comment (jkadamczyk) [inline src/helpers/buildWalletSections.js:210]: Why not use nullish coalescing here?

**Review by jkadamczyk — APPROVED** [review]
Looks good to me!
Left a couple of nits only

**Comment by osdnk** [comment]
@jkadamczyk regarding all your comments - no profound reason, I didn't change what was not necessary, but sure, I'll make it with nullish coalescing

**Review by brunobar79 — APPROVED** [review]
Added one minor comment, otherwise LGTM!

File: src/entities/tokens.ts [hunk 9]
```diff
@@ -63,6 +63,7 @@ export interface ParsedAddressAsset
     Partial<RainbowTokenOwnFields> {
   balance?: {
     amount?: string;
+    amountNum?: number;
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/entities/tokens.ts:66]: Unfortunately we need this exactly when we need to send eventing (max button)

File: src/helpers/savings.js [hunk 10]
```diff
@@ -28,8 +27,7 @@ const calculateCompoundInterestInDays = memoFn((principal, apr, days) => {
 });
 
 const formatSavingsAmount = memoFn(amount => {
```
Comment (terrysahaidak) [inline src/helpers/savings.js:29]: i think we don't need memo anymore

**Review by terrysahaidak — APPROVED** [review]
All good!

**Review by estebanmino — APPROVED** [review]
@osdnk let us know if you need someone from profiles to take a look, I did try this and it was working correctly after updating some methods

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Some issues from QA: 
- https://linear.app/rainbow/issue/RNBW-3899/[remove-bignumbersjs-3289]-android-gast-estimation-issues
- https://linear.app/rainbow/issue/RNBW-3900/[remove-bignumbersjs-3289]-error-on-expanding-savings-section
- https://linear.app/rainbow/issue/RNBW-3901/[remove-bignumbersjs-3289]-speed-up-and-cancel-renders-an-error
- https://linear.app/rainbow/issue/RNBW-3902/[remove-bignumbersjs-3289]-wc-approve-request-not-working
- https://linear.app/rainbow/issue/RNBW-3903/[remove-bignumbersjs-3289]-nothing-happens-when-attempting-deposit-and

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Just two outstanding issues from the bug fix QA:

**Issue 1:**

Issue 2 from RNBW-3899 is still occurring, seen here = http://recordit.co/c0wSylJXdX

**Issue 2:** 

The activity list shows as empty when I should have txn's populating on a fresh install of android and after clearing cache on iOS; seen here = https://recordit.co/Zqysx7U8z7

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Issue 1 = max base fee still changes when editing miner tip = http://recordit.co/qlSPrIq354

Issue 2 = fixed

**Review by ibrahimtaveras00 — APPROVED** [review]
I saw the last issue get fixed from @osdnk on both OS's, everything looks good 👍🏽

**Pull request merged**